CoreFoundation/String.subproj/CFString.c

```diff
@@ -1480,12 +1480,14 @@ CF_PRIVATE CFStringRef __CFStringCreateImmutableFunnel3(
                 useInlineData = true;
                 size = numBytes;
 
-                if (hasLengthByte || (encoding != kCFStringEncodingUnicode && __CFCanUseLengthByte(numBytes))) {
+                if (hasLengthByte) {
+                    useLengthByte = true;
+                    size += 1;
```

Comment:
It looks like you've reversed which condition gets the +1 size here? Shouldn't it be in !hasLengthByte?